internal/metrics/chat_completion_metrics.go

```diff
@@ -68,12 +68,12 @@ func (c *chatCompletion) SetBackend(backend filterapi.Backend) {
 }
 
 // RecordTokenUsage implements [ChatCompletion.RecordTokenUsage].
-func (c *chatCompletion) RecordTokenUsage(ctx context.Context, inputTokens, outputTokens, totalTokens uint32) {
-	attrs := []attribute.KeyValue{
+func (c *chatCompletion) RecordTokenUsage(ctx context.Context, inputTokens, outputTokens, totalTokens uint32, extraAttrs ...attribute.KeyValue) {
+	attrs := append([]attribute.KeyValue{
 		attribute.Key(genaiAttributeOperationName).String(genaiOperationChat),
 		attribute.Key(genaiAttributeSystemName).String(c.backend),
 		attribute.Key(genaiAttributeRequestModel).String(c.model),
-	}
+	}, extraAttrs...)
```

Comment:
can you pre allocate the capacity for the base slice ?

Reply:
Done